src/gatsby/src/templates/parkSubPage.js

```diff
@@ -44,23 +44,19 @@ export default function ParkSubPage({ data }) {
     useRef(null),
   ]
 
-  let pageSections = []
-  if (hasSections) {
-    let sectionIndex = 0
-    for (const s of sections) {
-      sectionIndex += 1
-      s.sectionIndex = sectionIndex
+  const pageSections = sections.map((section, sectionIndex) => {
+    sectionIndex += 1
+    section.sectionIndex = sectionIndex
```

Comment:
We generally want to avoid mutating parameters in functions like this; I didn't realize there was a "side effect" of changing `section.sectionIndex` when I suggested using a `.map` and I kind of want to suggest reverting it back to a normal loop now 😅 , but either way is fine. It's just that a lot of eslint configs would complain about [reassigning the parameters](https://eslint.org/docs/latest/rules/no-param-reassign).

How about instead of changing the `sectionIndex` parameter on this line:
```js
sectionIndex += 1
```
We just do the `+1` when we're building the returned object:
```diff
return {
--  sectionIndex: sectionIndex,
++  sectionIndex: sectionIndex + 1,
  ...
```
Then everything would still be fairly easy to follow.

Let me know if you have any questions!

Reply:
Thanks Duncan, and sorry I forgot to push the revert yesterday 😅 If I change the code below, the side menu item doesn't get highlighted when users scroll the screen, so I'll go with the initial change now but we can check this again later.

```
return {
--  sectionIndex: sectionIndex,
++  sectionIndex: sectionIndex + 1,
  ...
  ```